### PR TITLE
Many updates, fixes, and refactors

### PR DESCRIPTION
**Pull request by alexdiliberto**
* Ember CLI v2.14.2
* `babel-eslint`
* Node v4.x consumer support
* Headless chrome testing in Dev and CI
* Some refactors + cleanup
* `start:inspect` NPM script
* Cleaner code block whitespace control/formatting in `templates/components/themed-syntax.hbs`

**Before (without whitespace control):**
screen shot 2017-08-05 at 3 52 23 pm

**After (with whitespace control):**
**notice the less space on top/bottom**
screen shot 2017-08-05 at 3 52 32 pm

* Proper intent handling

```js
{{#themed-syntax lang="bash"}}
      function foo() {
    return 'bar';
  }

    console.log('baz');
{{/themed-syntax}}
```

**Before:**
screen shot 2017-08-05 at 6 22 09 pm

**After:** 
screen shot 2017-08-05 at 6 20 59 pm

### CODE REVIEW

**Comment by alexdiliberto** [comment]
@crodriguez1a Once this is is signed off and merged, please cut a new minor version release `0.1.0`

**Comment by crodriguez1a** [comment]
👍🏽 @alexDiLiberto, I'll review today

File: README.md [hunk 1]
```diff
@@ -6,6 +6,10 @@ An out-of-the-box solution for syntax highlighting with a couple of themes inclu
 
 * `ember install ember-themed-syntax`
 
+## Demo and Documentation
+
+http://demos.evolutionaryapps.com/EmberThemedSyntax
```
Comment (crodriguez1a) [inline README.md:11]: @alexdiliberto We should probably update this to be gh-pages instead.

File: addon/components/themed-syntax.js [hunk 2]
```diff
@@ -103,44 +105,41 @@ export default Ember.Component.extend({
...
-      let syntax = highlight(raw, {
-        hljs,
-        lang,
-        start: Number(numbers)
-      });
-
-      // Output formatted
-      set(this, 'code', syntax);
+    // Get the trimmed/buffered raw text
+    let raw = `${this.$().find('.code').text()}`;
```
Comment (crodriguez1a) [inline addon/components/themed-syntax.js:118]: @alexdiliberto We can also do `import $ from "jquery"`

File: addon/templates/components/themed-syntax.hbs [hunk 3]
```diff
@@ -1,7 +1,5 @@
...
-    {{yield}}
-  {{else}}
-  {{{code}}}
-  {{/unless}}
-</code>
+{{!
+  NOTE: Don't update this layout/spacing/tabbing
+  This format allows for the best looking whitespace in codeblocks
+}}
+<code class="code hljs {{if isDark 'dark'}}">{{#unless code}}{{yield}}{{else}}  {{{code}}}{{/unless}}</code>
```
Comment (crodriguez1a) [inline addon/templates/components/themed-syntax.hbs:5]: @alexdiliberto 👍

**Review by crodriguez1a — APPROVED** [review]

**Comment by crodriguez1a** [comment]
@alexdiliberto Thanks for the support on this man! I haven't had a whole lot of extra time to keep this up-to-date.

**Comment by crodriguez1a** [comment]
@alexdiliberto I added you as an NPM collaborator as well. I setup a draft release. Feel free to publish.

**Pull request merged**